src/util/namespace.h

```diff
@@ -84,8 +84,14 @@ class namespacet:public namespace_baset
 
   using namespace_baset::lookup;
 
-  // these do the actual lookup
+
+  /// Searches for a symbol named \p name in `symbol_table1` and returns a
+  /// pointer to it in output parameter \p symbol when found. When no symbol is
+  /// found, it performs the same lookup in `symbol_table2`.
+  /// \return False iff the requested symbols is found in one of the two tables.
   virtual bool lookup(const irep_idt &name, const symbolt *&symbol) const;
+
+  // this and lookup above do the actual lookup
```

Comment:
Note that `git blame` will associate that change with your name, so be very careful about that sort of copying.

I didn't know anything about `get_max` either, but from reading the code it seems to do: for all symbols the name of which have a prefix `prefix` and an assuming this prefix is followed by an integer, return the maximum such integer. The intended use case is finding the next available symbol name for a sequence of auto-generated symbols.